apps/platform/src/app/(main)/(project)/[workspace]/[project]/@environment/page.tsx

```diff
@@ -39,24 +39,32 @@ function EnvironmentPage(): React.JSX.Element {
   )
 
   useEffect(() => {
-    if (!selectedProject) return 
-    setLoading(true)
+  if(!selectedProject) return
+   handleEnvironmentsFetch();
+  }, [selectedProject, getAllEnvironmentsOfProject, setEnvironments])
+
+  const handleEnvironmentsFetch = (newPage = 0) => {
+    if (!selectedProject) return
+    setIsLoading(true)
     getAllEnvironmentsOfProject()
       .then(({ data, success }) => {
         if (success && data) {
-          const newData = page === 0 ? data.items : [...environments, ...data.items];
-          setEnvironments(newData);
-          if (newData.length >= data.metadata.totalCount) setHasMore(false);
+          const newData = newPage === 0 ? data.items : [...environments, ...data.items]
+          if(newPage == 0 && page !== 0) setPage(0);
+          setEnvironments(newData)
+          if (newData.length >= data.metadata.totalCount) setHasMore(false)
```

Comment:
you can check for `data.metadata.next`. if its null, you can set `hasMore` to false